src/test/CustomCurveHook.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {Hooks} from "../libraries/Hooks.sol";
+import {SafeCast} from "../libraries/SafeCast.sol";
+import {IHooks} from "../interfaces/IHooks.sol";
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+import {PoolKey} from "../types/PoolKey.sol";
+import {BalanceDelta, toBalanceDelta} from "../types/BalanceDelta.sol";
+import {Currency} from "../types/Currency.sol";
+import {BaseTestHooks} from "./BaseTestHooks.sol";
+import {IERC20Minimal} from "../interfaces/external/IERC20Minimal.sol";
+import {CurrencyLibrary, Currency} from "../types/Currency.sol";
+
+contract CustomCurveHook is BaseTestHooks {
+    using Hooks for IHooks;
+    using CurrencyLibrary for Currency;
+
+    error AddLiquidityDirectToHook();
+
+    IPoolManager immutable manager;
+
+    constructor(IPoolManager _manager) {
+        manager = _manager;
+    }
+
+    modifier onlyPoolManager() {
+        require(msg.sender == address(manager));
+        _;
+    }
+
+    function beforeSwap(
+        address, /* sender **/
+        PoolKey calldata key,
+        IPoolManager.SwapParams calldata params,
+        bytes calldata /* hookData **/
+    ) external override onlyPoolManager returns (bytes4, int128) {
+        (Currency inputCurrency, Currency outputCurrency, uint256 amount) = _getInputOutputAndAmount(key, params);
+
+        // this "custom curve" is a line, 1-1
+        // take the full input amount, and give the full output amount
+        manager.take(inputCurrency, address(this), amount);
+        _settle(outputCurrency, amount);
+
+        // return -amountSpecified to no-op the concentrated liquidity swap
```

Comment:
I think the sign matters here depending on the type of swap

Hook would return -amountSpecified for an exactOutput swap but would have to return +amountSpecified for an exactInput swap.

Reply:
but its always the inverse of amountSpecified.
if amountSpecified=-100, it does indeed need to return a positive number, but thats `-amountSpecified` which is `-(-100)` which is `+100`
if amountSpecified=100, it needs to return a negative number, which is `-amountSpecified` which is `-100`

so it always returns -amountSpecified

Reply:
ah yes yes totally i guess i was forgetting that amountSpecified is already signed idk... i forget writing that comment hahaha 😆